pkg/worker/worker.go

```diff
@@ -126,6 +126,12 @@ func (w *worker) SetWorkerFunc(workerFunc func(interface{}) (ctrl.Result, error)
 
 // SubmitJob adds the job to the rate limited queue
 func (w *worker) SubmitJob(job interface{}) {
+	// in theory, only health check endpoint should send a nil job to test periodically
+	if job == nil {
```

Comment:
nit: maybe good to check if it is the health check endpoint here.

Reply:
I don't think there us an easy way to check that. We will have to do either of
- mock the parameter to reflecting caller identity. But this will be no difference from current `nil`
- use runtime package to check caller frames. I think this adds unnecessary complexity.